### ISSUE TITLE
Expose Gradle configuration via properties

### ISSUE THREAD

**Issue by jonapoul** [detekt/detekt#9117]
I'm proposing adding a more widespread ability to configure Detekt's Gradle setup via properties, for example:

```sh
gradle detektMain \
  -Pdetekt.failOnSeverity=Info \
  -Pdetekt.buildUponDefaultConfig=false
# etc etc
```

and/or in the gradle.properties file:

```properties
detekt.failOnSeverity=Info
detekt.buildUponDefaultConfig=false
```

This would allow build authors of larger/more complex Gradle projects to more quickly change configs and validate their setups without having to invalidate everything and have to sit through the full Gradle rebuild.

This would only really be relevant for primitive values (bools, enums, strings). We could also add files/directories, but need to be explicit about what they're defined relative to. If they're too controversial the filepaths could be skipped for the initial impl.

## Expected Behavior
Personally speaking, I'd like to see the property take priority over the extension value, so setting `detekt.someProperty` in my properties file or CLI call will override the value set in the Gradle script:

```kotlin
val someProperty: Property<SomeProperty> = providers
    .gradleProperty("detekt.someProperty")
    .map { string -> parseSomeProperty(string) }
    .orElse(detektExtension.someProperty)
```

Although in my experience plugin authors tend to go the other way around, so the property will only take effect if there's no explicit setter in the build scripts:

```kotlin
val someProperty: Property<SomeProperty> = detektExtension.someProperty
    .orElse(providers.gradleProperty("detekt.someProperty"))
    .map { string -> parseSomeProperty(string) }
```

I'm not a maintainer so that's not my decision, but the first would be my personal preference of the two.

## Current Behavior
Can only use the build script interface for the majority of options, except `detekt.use.worker.api` and `detekt-dry-run` properties. Neither of these two currently have a config interface via the `DetektExtension`. These two properties use different property naming/scoping conventions, so there might need to be some form of "migration" to a more standardised convention? E.g. `detekt.someProperty`

## Context
I sometimes have to do some fiddling with Detekt configurations. This might involve changing a flag or adding something to a list and running a build to see how it affects the rest of the (large) project. This invalidates the cache because the config is in a central build-logic module.

If I could instead do the same via one Gradle property, I'd maintain the wider cache, only invalidating the property (and any dependent tasks) that I'm reconfiguring.

This is purely an extra interface for configuration, no behavioural change to any rules.

Thoughts?

**Comment by BraisGabin** [comment]
I would not be against some of them (I'm looking at you, autocorrect). But I don't know where to draw the line. 

And also, is something stopping you to configure that yourself in the build src? You could set the value reading it from a property and adding a default if there is none.

**Comment by jonapoul** [comment]
> But I don't know where to draw the line.

Personally I would allow primitives, enums and maybe string lists via CSV, but not files (unless maybe it's relative to the root project?). Project-specific config via properties will be a no-go, the best practice `providers.gradleProperty` API doesn't support that AFAIK.

> And also, is something stopping you to configure that yourself in the build src?

No, this is just a bit of extra configuration flexibility.

**Comment by 3flex** [comment]
> this is just a bit of extra configuration flexibility

I'd prefer to close this, as there's a simple enough way to configure this for builds where it's required. If we add first party support then it can result in quite a lot of extra code, and as mentioned above, it's not clear where it makes sense to draw the line if we only add this for a selection of parameters.
